### DISCUSSION TITLE
Reference Depths - need to add 6?

### DISCUSSION THREAD

**Discussion by gmcnicol** [oapi-codegen/oapi-codegen#637] [Q&A]
I'm trying to generate a client from the following YAML: https://developer.experian.com/system/files/2021-06/experian-cato-api-v2.1.0-openapi-global-developer-portal-1.yaml

They have bundled all the individual field types under a FieldSchemas object (line 725). Am I guessing this is valid in OpenAPI? 

However, this has hit the checks in codegen/utils.go refPathToGoType (line 220). 

Should I raise an issue? Should I raise a PR? 

Thanks for oapi-codegen - brilliant tool.

**Comment by gmcnicol** [comment]
I am digging further into the YAML. Each model's properties are a $ref to a FieldSchemas property. These properties do not get correctly created or correctly processed because they are primitives.

Determining the type name for the field is nieve.

**Reply by realCommitment** [reply]
Hey, have you found a solution to this? I have encountered the same issue.
